hist/hist/inc/TFormula.h

```diff
@@ -97,13 +97,16 @@ class TFormula : public TNamed
    Bool_t            fLazyInitialization = kFALSE; ///<! Transient flag to control lazy initialization (needed for reading from files)
    std::unique_ptr<TMethodCall> fMethod;           ///<! Pointer to methodcall
    std::unique_ptr<TMethodCall> fGradMethod;       ///<! Pointer to a methodcall
+   std::unique_ptr<TMethodCall> fHessMethod;       ///<! Pointer to a methodcall
```

Comment:
Apologies for this question but I cannot see the answer on a quick glance: what do we need both `fHessMethod` and `fHessFuncPtr`?

Reply:
If you take a look at line `3330` of `TFomula.cxx` (I am unable to get the link to the line) tyou will be able to see that `fHessFuncPtr` is used to call the generated hessian. Line `3279` of the same file is where the fHessMethod is used to store the TMethodCall that has the same prototype as the generated hessian. Also I have made a type in line `140` of `TFormula.h` file. It should be `fHessMethod` instead of `fGradMethod`

Reply:
Thanks! IIUC you're saying `fHessFuncPtr` should be sufficient, as `fHessMethod` is only set, and its only reading is in `HasHessianGenerationFailed()` which - again, IIUC - could as well use `fHessFuncPtr`?

Reply:
Yeah